include/nbl/ext/ScreenShot/ScreenShot.h

```diff
@@ -18,228 +15,108 @@ namespace nbl
-
-						state.extent = regionWithMipMap->getExtent();
-						state.layerCount = regionWithMipMap->imageSubresource.layerCount;
-						state.inMipLevel = regionWithMipMap->imageSubresource.mipLevel;
-						state.outMipLevel = regionWithMipMap->imageSubresource.mipLevel;
-
-						const bool ok = convertFilter.execute(&state);
-						assert(ok);
+						auto newCreationParams = cpuNewImage->getCreationParameters();
+
+						asset::ICPUImageView::SCreationParams viewParams;
+						viewParams.flags = static_cast<asset::ICPUImageView::E_CREATE_FLAGS>(0u);
+						viewParams.image = cpuNewImage;
+						viewParams.format = newCreationParams.format;
+						viewParams.viewType = asset::ICPUImageView::ET_2D;
+						viewParams.subresourceRange.baseArrayLayer = 0u;
+						viewParams.subresourceRange.layerCount = newCreationParams.arrayLayers;
+						viewParams.subresourceRange.baseMipLevel = 0u;
+						viewParams.subresourceRange.levelCount = newCreationParams.mipLevels;
+
+						cpuImageView = asset::ICPUImageView::create(std::move(viewParams));
 					}
 				}
-				else
-					convertedImage = image;
-				auto newCreationParams = convertedImage->getCreationParameters();
-				
-				asset::ICPUImageView::SCreationParams viewParams;
-				viewParams.flags = static_cast<asset::ICPUImageView::E_CREATE_FLAGS>(0u);
-				viewParams.image = convertedImage;
-				viewParams.format = newCreationParams.format;
-				viewParams.viewType = asset::ICPUImageView::ET_2D;
-				viewParams.subresourceRange.baseArrayLayer = 0u;
-				viewParams.subresourceRange.layerCount = newCreationParams.arrayLayers;
-				viewParams.subresourceRange.baseMipLevel = 0u;
-				viewParams.subresourceRange.levelCount = newCreationParams.mipLevels;
-
-				auto imageView = asset::ICPUImageView::create(std::move(viewParams));
-
-				auto tryToWrite = [&](asset::IAsset* asset)
-				{
-					asset::IAssetWriter::SAssetWriteParams wparams(asset);
-					return assetManager->writeAsset(outFileName, wparams);
-				};
-
-				bool status = tryToWrite(convertedImage.get());
-				if (!status)
-					status = tryToWrite(imageView.get());
 
-				destinationBoundMemory->unmapMemory();
-				return status;
+				nbl::core::smart_refctd_ptr<nbl::asset::IAssetManager> assetManager;
+				{
+					nbl::core::smart_refctd_ptr<nbl::io::IFileSystem> fileSystem = nbl::core::make_smart_refctd_ptr<nbl::io::CFileSystem>("");
+					assetManager = core::make_smart_refctd_ptr<nbl::asset::IAssetManager>(std::move(fileSystem));
+				}
 
-			}
-			inline bool createScreenShot(core::smart_refctd_ptr<IrrlichtDevice> device, const video::IGPUImageView* gpuImageView, const std::string& outFileName)
-			{
-				auto driver = device->getVideoDriver();
-				auto assetManager = device->getAssetManager();
-				return createScreenShot(driver, assetManager, gpuImageView, outFileName);
+				nbl::asset::IAssetWriter::SAssetWriteParams writeParams(cpuImageView.get());
+				return assetManager->writeAsset(outFileName, writeParams);
```

Comment:
can you restore the separation of responsibilities (one overload to make an ICPUImageView, one to call the first and save it to file)